### PR TITLE
Jordanjones243/bib full width

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** #86, #88

## Summary:

Added missing dev dependencies for testing to work, used the new `common` attribute to replace other attributes in `auro-dropdown`, and added the `matchWidth` attribute.

Added support for `auro-menu`'s optional checkmark.

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

**Review by blackfalcon — DISMISSED** [review]

**Comment by blackfalcon** [comment]
@jordanjones243 good job on the update!

**Review by blackfalcon — CHANGES_REQUESTED** [review]
Couple of things. 

1. The API docs are autogenerated and require you to run `npm run build` to update
2. Tests are below threshold

```
Code coverage: 89.74 %
Coverage for branches failed with 69.44 % compared to configured 70 %
View full coverage report at coverage/lcov-report/index.html

Finished running tests in 1.8s, failed to meet coverage threshold.
```

File: src/auro-select.js [hunk 1]
```diff
@@ -244,6 +251,22 @@ class AuroSelect extends LitElement {
...
+        this.menus[index].setAttribute('nocheckmark', '');
+      }
+    }
+
+    if (this.checkmark) {
+      this.querySelectorAll('auro-menu').forEach((menu) => menu.removeAttribute('nocheckmark'));
+    } else {
+      this.querySelectorAll('auro-menu').forEach((menu) => menu.setAttribute('nocheckmark', ''));
+    }
+
```
Comment (jason-capsule42) [inline src/auro-select.js:269]: I believe this code should be removed because we are now doing it in the for loop above at line 254.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

File: src/auro-select.js (hunk 1 above)
Comment (jordanjones243) [reply src/auro-select.js]: Duplicate code removed.

**Review by jason-capsule42 — APPROVED** [review]

**Review by blackfalcon — CHANGES_REQUESTED** [review]
This PR has breaking changes that are not accounted for. There is work completed that is not in the scope of the issue attached. 

Also, the updated API docs are not included in the PR.

**Comment by jason-capsule42** [comment]
> This PR has breaking changes that are not accounted for. There is work completed that is not in the scope of the issue attached.
> 
> Also, the updated API docs are not included in the PR.

The PR has been updated with the additional ticket for the checkmark support that was added in auro-menu.

I can have @jordanjones243 update his commit to identify the checkmark change as `breaking` but I am not sure I agree it is that. This change makes auro-select work as intended by default.

I do see the API docs are missing as well - @jordanjones243 all you need to do here is `npm run build`. You should see `api.md` and `apiExamples.md` files updated. I know we ran the build multiple times while we were pair programming. If you run that again and don't see those files changed let me know and I will help you out.

When you commit those API doc updates we can just add the `BREAKING CHANGE` text to the commit. Let me know if you have any questions about that and we can jump in a call whenever you are ready.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]
`api.md` and `apiExamples.md` files need to be updated by running `npm run build`.

**Comment by blackfalcon** [comment]
> I can have @jordanjones243 update his commit to identify the checkmark change as breaking but I am not sure I agree it is that. This change makes auro-select work as intended by default.

I am not following. The current version as displayed in https://auro.alaskaair.com/components/auro/select, the checkmark does not require an attribute to be used. 

This update changes that expectation. That is a breaking change.

**Comment by blackfalcon** [comment]
Given that auro-combobox was just released with `checkmark` as the API model to hide/show the checkmark UI, we should move this in the same direction. But this IS a breaking change and the commit needs to reflect that.

File: src/auro-select.js [hunk 2]
```diff
@@ -69,6 +72,10 @@ class AuroSelect extends LitElement {
         type: Boolean,
         reflect: true
       },
+      checkmark: {
+        type: Boolean,
+        reflect: true
+      },
```
Comment (blackfalcon) [inline src/auro-select.js:78]: To remove the necessity of a MAJOR release for this update that is inconsistent with how the other `noCheckmark` attributes are working with similar UIs, we should reverse this logic to use `noCheckmark` for a MINOR release. 

There is work to be addressed in the upcoming months regarding forms and consistency between them where we can revisit this feature across all similar UI elements and if decided to have major changes we will release MAJOR versions of all form elements at the same time.

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: demo/apiExamples.md [hunk 3]
```diff
@@ -16,7 +16,7 @@ The auro-select element is a wrapper for auro-dropdown and auro-menu to create a
 
 | Property         | Attribute        | Type      | Default                | Description                                      |
 |------------------|------------------|-----------|------------------------|--------------------------------------------------|
-| [checkmark](#checkmark)      | `checkmark`      | `Boolean` |                        | When attribute is present auro-menu will apply checkmarks to selected options. |
+| [checkmark](#checkmark)      | `checkmark`      | `Boolean` | true                   | When attribute is present auro-menu will apply checkmarks to selected options. |
```
Comment (blackfalcon) [inline demo/apiExamples.md:19]: I thnk you missed updating the JSDocs comments in the auro-select.js file.

File: docs/api.md [hunk 4]
```diff
@@ -6,7 +6,7 @@ The auro-select element is a wrapper for auro-dropdown and auro-menu to create a
 
 | Property         | Attribute        | Type      | Default                | Description                                      |
 |------------------|------------------|-----------|------------------------|--------------------------------------------------|
-| `checkmark`      | `checkmark`      | `Boolean` |                        | When attribute is present auro-menu will apply checkmarks to selected options. |
+| `checkmark`      | `checkmark`      | `Boolean` | true                   | When attribute is present auro-menu will apply checkmarks to selected options. |
```
Comment (blackfalcon) [inline docs/api.md:9]: Same.

**Review by blackfalcon — CHANGES_REQUESTED** [review]

**Comment by blackfalcon** [comment]
The last merging of `v2.5-rc` into the working feature branch somehow meshed the last three releases into this feature branch and there are still conflicts with the `v2.5-rc` branch that cannot be undone. 

The recommendation is to close this branch and PR and redo the work.

**Pull request closed without merging**